### DISCUSSION TITLE
How to avoid DROP TABLE when I just changed / added some parameters to a table?

### DISCUSSION THREAD

**Discussion by deemeetree** [prisma/prisma#11275] [Q&A] [answered]
Hello, 

I have a Prisma / Nest.Js app using PostgreSQL.

We have updated our schema and I ran, as suggested, `prisma generate` and then `prisma migrate dev --create-only`. 

It generates a migration file, however, which DROPS my tables, when in fact the only thing I changed was to add some new properties to them and to update some names of those properties.

Why does this happen? How can I update my DB schema with Prisma without removing my DB? 

I understand that I can make a copy of it and then recreate the data using seeding, but I don't like this option. It just seems so inconvenient to have to do this every time I have to change some small thing in the DB schema. 

So what is another way to solve this problem? Can somebody explain it to me, please? Thanks!

**Comment by austincrim** [comment, marked as answer]
Hey @deemeetree 👋 ,

Check out our [docs page on renaming fields.](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/customizing-migrations#renaming-a-field). 

In short, Prisma can't tell the difference between renaming a field in the schema and deleting a field and adding an entirely different one . You can avoid dropping tables by manually editing your generated `.sql` file and changing the `DROP` commands to rename commands (this varies a bit by database provider).

Let me know if I can clarify anything!

**Reply by deemeetree** [reply]
Ok thank you, but then what's the point of having a migration feature in Prisma if it simply doesn't work and I have to do everything manually? I mean, renaming a field or adding a property is such a simple task — why can't it be solved programmatically? I thought this was part of Prisma's original task, no?

**Reply by janpio** [reply]
There is technically no way how Prisma can differentiate between you wanting to a) rename a field or b) remove a field and add a new one. They both look like the same thing if you compare before and after. All heuristics we could apply on the Prisma Schema file level, could have false positives and would require the user to still check - so we default to one of the obvious choices and document that. (If you have a solution for this problem, please let us know though - we are very happy to improve this!)

We have a feature request to improve this situation though via an interactive process: https://github.com/prisma/prisma/issues/7790

You can add a property and Prisma will generate the correct SQL for that case though - that is the main use case indeed.

**Reply by monapasan** [reply]
What about adding a flag --rename that will prioritize renaming over removal when creating a migration?

**Reply by monapasan** [reply]
when I try to rename a value in a enum it does the following:
1. create new type containing newly renamed field
2. drop all default values of tables that use this enum
3. change tables to use the new enum type
4. renaming old enum type and droping it
5. use default values in the newly create enum type

I am not sure but this is look to me quite a unoptimisitc way of creating a migration.
This can be done in one step by simply having this query:
`ALTER TYPE enum_type RENAME VALUE 'old_value' TO 'new_value'
which will be one nest instead of 6 + without any data lost.
@janpio Could you please point me out where this logic is located 🙏

**Reply by monapasan** [reply]
Sorry, actually current prisma migrate produces 9 lines instead of potential 1 for this change. Even just adding new enum value and removing old one would be 2 lines (even though this will still result in the data lost)

**Reply by janpio** [reply]
Please open an issue if you think this behavior is suboptimal, and provide all the information the template asks for. Thanks.

**Reply by swiser-gus** [reply]
Thoughts on implementing attribute annotations and classes? I come from a php background and I know laravel does it this way. Have you guys explored this option? If so can you share your thoughts?

**Comment by Frosty21** [comment]
The best way I've found is doing it this way for Table which should be similar to renaming parameters  https://stackoverflow.com/a/71616425
not ideal but it works. I can see that this could be annoying but I hope the community can take initiative to make a PR to solve this small issue when compared to other features needed.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
